draft-ietf-ice-pac.xml

```diff
@@ -176,18 +176,21 @@
         needs to wait enough time to allow peer-reflexive candidates to be
         discovered. Accordingly, when a full ICE implementation begins its
         ICE processing, as described in <xref target="RFC8445" />, Section
-        6.1, it MUST set a timer, and its duration SHOULD be
-        equal to the agent's connectivity check transaction timeout.
+        6.1, it MUST set a timer, and its duration MUST be sufficient to allow
+        all peer-reflexive candidates to be discovered before firing.
       </t>
```

Comment:
How can you test such "MUST"? We can only say something like "MUST be sufficient in order to the agent to assume that all peer-reflexive candidates are discovered before firing", and even that would be diffuse.

Why not simply say:

"The RECOMMENDED duration for the timer is equal to the agent's
connectivity check transaction timeout, as that is considered to be
sufficient to allow all peer-reflexive candidates to be discovered before
the timer times out."

- I don't we need to talk about retransmissions, as they are part of the transaction
- I suggest to say "times out" instead of "firing".

Reply:
@pthatcherg suggested this, otherwise the whole document has no MUST-strength behavior. See #9. 

I do think it's important to talk about retransmissions, otherwise it's unclear to a casual reader whether they are included.

Reply:
We do say MUST set a timer, and later we recommend a value for the timer. In my PR I also say MUST NOT declare ICE failure while the timer is still running.

Reply:
@pthatcherg, wdyt?

Reply:
In the interest of progressing this PR, I removed the ambiguous MUST. PTAL.

Reply:
I think "to enforce a minimum amount of time for ICE to run" needs to be clarified, as it is ok to conclude ICE before the minimum amount of time has elapsed, if the ICE agent is able to nominate pairs.

We need to say that we enforce a minimun amount of time for ICE to run before we consider a checklist Failed, or something like that.

Reply:
PTAL

Reply:
@cdh4u, are you good with this?

Reply:
As far as the actual change above, I am fine with the text. I guess we should use capital F for 'Failure', though.

Just to clarify one thing: the text says that the timer is SET when ICE processing starts, but according to #15 it STARTS a little later (currently once the ICE credentials have been exchanged). So, I assume that "SET" only means that the agent determines a timer value - it does not yet start the timer, right?


However, we need to make sure this text and the text in #15 is aligned. In one place we say that the timer is set when ICE processing starts (according to section 6.1 if 8445, it starts already when the role is determined), while in #15 we talk about starting the timer when the ICE credentials have been exchanged.

The text above says that the timer starts the timer when ICE processing starts (as defined in section 6.1 of 8445), and the text in #15 currently says that the timer starts once the ICE credentials have been exchanged. When reading section 6.1 of 8445, I think "ICE processing" starts *before* the credentials have been exchanged.

Reply:
No, I think everyone would agree that setting and starting a timer are the same thing. (e.g. JS setTimeout)

Reply:
also, I don't think we should use 'Failure', since that isn't a defined ICE state, and I don't think 'failure' is ambiguous here.

Reply:
> No, I think it is generally understood that setting and starting a timer are the same thing. (e.g. JS setTimeout)

Let's use the same word in both places then, to avoid confusion.

Reply:
> also, I don't think we should use 'Failure', since that isn't a defined ICE state, and I don't think 'failure' is ambiguous here.

Fair enough. However, "declaring failure" is not used in 8445, so perhaps saying something like:

"before declaring ICE failure (i.e., setting the ICE state to Failed)."

Reply:
Honestly, I think this is already implicit because the current text explicitly talks about ICE running for a minimum amount of time. The abstract also contains similar text.

Reply:
Ok, so let's keep the current suggestion for now. Once we have a new version of the whole draft it will be easier to get an overview, and see whether something still needs to be clarified.